.github/workflows/ci-complete.yml

```diff
@@ -21,14 +21,30 @@ on:
     types:
       - completed
 
+run-name: Build Scans for ${{ github.event.workflow_run.display_title}}
+
+# This workflow runs after the completion of the CI workflow triggered on a "pull_request" event.
+# The "pull_request" event type is run in an unprivileged context without access to the repository
+# secrets. This means that PRs from public forks cannot publish Gradle Build Scans or modify the
+# PR contents.
+#
+# This "workflow_run" triggered workflow is run in a privileged context and so does have access to
+# the repository secrets. Here we can download the build scan files produced by a PR and publish
+# them to ge.apache.org.
+#
+# If we need to do things like comment on, label, or otherwise modify PRs from public forks. This
+# workflow is the place to do it. PR number is ${{ github.event.workflow_run.pull_requests[0].number }}
+
 jobs:
   upload-build-scan:
-    if: ${{ github.event.workflow_run.head_repository.full_name != 'apache/kafka' }}
+    # Skip this workflow if CI was run for anything other than "pull_request" (like "push").
+    # Also skip this workflow if the PR was from apache/kafka. Those will have already published the build scan in CI.
+    if: ${{ github.event.workflow_run.event == 'pull_request' && github.event.workflow_run.head_repository.full_name != 'apache/kafka' }}
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        java: [ 21, 17, 11, 8 ]
+        java: [ 17, 11 ]
```

Comment:
Could you please comment to https://github.com/apache/kafka/blob/trunk/.github/workflows/build.yml#L88 to remind that we need to keep the JDK consistency.